### ISSUE TITLE
Detailed Error Handling 

### ISSUE THREAD

**Issue by kongweiying2** [fwouts/previewjs#1382]
### Describe the bug

Sometimes Preview.js describes a generic error message that implies it's responsible for the error when in fact the error isn't due to Preview.js at all. In this example, it throws a module import error, but in actuality, the real error is shown in the output logs for the Preview.js extension.

```
Unable to render RichTextCodeExample
While Preview.js can work out of the box for simple components, you may need a custom configuration to render components that depend on:

a specific state or context (see [Wrapping components](https://previewjs.com/docs/config/wrapper))
modules imported via [aliases](https://previewjs.com/docs/config/aliases)
[static assets](https://previewjs.com/docs/config/static-assets) or [SVGR](https://previewjs.com/docs/config/svgr)
Please see the logs below:

[12:59:03] TypeError: Failed to fetch dynamically imported module: http://localhost:3140/preview/src/view/content/RichText.tsx
If you get stuck, do not hesitate to ask for help on [Discord](https://discord.previewjs.com/) or [GitHub](https://github.com/fwouts/previewjs/issues)!
```
As shown here!
```
12:49:03 PM [vite] page reload src/view/content/RichText.tsx
Failed to load url /src/view/content/styles.scss (resolved id: /src/view/content/styles.scss). Does the file exist?
```

### Reproduction

1. Create a simple React component that imports a './styles.scss' sass CSS file
2. Import this React component in another simple React component.
3. Open Preview.js on the parent React component.
4. Remove the styles.scss file that is being imported, from the directory.

### Preview.js version

v1.17.4

### Framework

"react": "16.14.0", "react-dom": "16.14.0"

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 305.94 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
Binaries:
    Node: 18.8.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.18.0 - /opt/homebrew/bin/npm
IDE:
    VSCode: 1.74.3
Browsers:
    Safari: 16.2


Also using Arc 0.85.0
```

### Used Package Manager

yarn

### Extension logs (useful for crashes)

```shell
1:01:56 PM [vite] page reload src/view/content/RichText.tsx
1:01:56 PM [vite] page reload src/view/content/RichText.tsx
Failed to load url /src/view/content/styles.scss (resolved id: /src/view/content/styles.scss). Does the file exist?
Failed to load url /src/view/content/styles.scss (resolved id: /src/view/content/styles.scss). Does the file exist?
New dependencies were detected. Restarting...
Preview.js Server stopped.
Preview.js Server running at http://localhost:3140.
1:02:02 PM [vite] page reload src/view/content/RichText.tsx
```

**Comment by fwouts** [comment]
Thanks for the bug report! It's always tricky for Preview.js to know if it's its own fault when there is an error, but this seems like a case where we could surface it much better 😊

**Comment by kongweiying2** [comment]
I do remember you mentioning that error handling is a tricky area, I guess the solution is harder than it seems!

**Comment by baditaflorin** [comment]
I would second this with more precise timestamping. 
I sometimes need to select a piece of text and re-run previewjs to make sure that now I am seeing the latest updates and not the log from 5 minutes ago. 

```
No workspace root detected from /static/asset-manifest.json
Preview.js Server stopped.
Warning: encountered some TypeScript error(s) while resolving types.
This may be safe to ignore, but could help point out why some components are not detected by Preview.js.
Cannot find name '____________'.
Preview.js Server running at http://localhost:3140.
No workspace root detected from /static/asset-manifest.json
Preview.js Server stopped.
```

**Comment by baditaflorin** [comment]
See my example here where Previewjs first loads for around 2 seconds and then it disappears 
<img width="939" alt="image" src="https://user-images.githubusercontent.com/2330999/222095515-367f6edf-7e57-4c1b-9afd-3ddb47a16e0d.png">
I have selected the text to make sure that the Preview.js Server running at http://localhost:3140. message is the latest one. 
After 2 seconds It crash but I cannot find anything that can help me pinpoint why this is happening. 
 <img width="1010" alt="image" src="https://user-images.githubusercontent.com/2330999/222095271-31fab333-9cdc-4519-8dbd-e99a5cea2ba4.png">

If I try 10 times to re-open a function I may get 

Preview.js Server running at http://localhost:3140.
Preview.js Server stopped.

And this is it. Nothing more gets logged in main, window, extension host or anywhere else. 
It's virtually impossible to debug in this circumstances.

**Comment by fwouts** [comment]
This should be fixed in v1.19.3, which just shipped :)
